Robot_Adapter/CRUD/Update/Panels.cs

```diff
@@ -0,0 +1,90 @@
+/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2020, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.oM.Structure.Elements;
+using BH.oM.Structure.SurfaceProperties;
+using RobotOM;
+using System.Collections.Generic;
+using BH.oM.Adapter;
+using System.Linq;
+
+namespace BH.Adapter.Robot
+{
+    public partial class RobotAdapter
+    {
+        /***************************************************/
+        /****           Protected Methods               ****/
+        /***************************************************/
+
+        protected override bool UpdateOnly<T>(IEnumerable<T> panels, string tag = "", ActionConfig actionConfig = null)
```

Comment:
Would not override this method, but just change it do an Update method that then can be dynamically called by the IUpdate mehtod. (similar principle as what we do with the Create)

By overriding this method UpdateOnly will only work for panels, but no other objects.

Reply:
Sorry missed that one... was still getting to grips with new adapter. Updated now and reran just to check - all ok. Have you seen my comments in message above? Line 165 and 193 in this cs file are relevant. I'm not sure it's 'correct' to be calling update on properties of dependent objects, and if it is, I don't think this approach is sustainable (and will have a big performance hit as it scales).